test/integration/gems_test.rb

```diff
@@ -155,22 +155,61 @@ class GemsSystemTest < SystemTest
     assert page.has_no_selector?(".github-btn")
   end
 
-  test "does show version that introduced mfa requirement" do
+  test "shows both blocks if latest AND viewed version require MFA" do
     @version.update_attribute :metadata, { "rubygems_mfa_required" => "true" }
+    create(:version, :mfa_required, rubygem: @rubygem, number: "0.1.1")
 
-    visit rubygem_path(@rubygem)
+    visit rubygem_version_path(@rubygem, "0.1.1")
 
-    assert page.has_content? "true"
+    assert page.has_content? "New versions require MFA"
+    assert page.has_content? "Version published with MFA"
   end
 
-  test "does show correct version that introduced mfa requirement" do
+  test "shows 'new' block only if latest requires MFA but viewed version doesn't" do
     @version.update_attribute :metadata, { "rubygems_mfa_required" => "true" }
-    create(:version, rubygem: @rubygem, number: "2.2.2")
-    create(:version, :mfa_required, rubygem: @rubygem, number: "3.3.3")
-    create(:version, :mfa_required, rubygem: @rubygem, number: "4.4.4")
+    create(:version, rubygem: @rubygem, number: "0.1.1")
+
+    visit rubygem_version_path(@rubygem, "0.1.1")
+
+    assert page.has_content? "New versions require MFA"
+    refute page.has_content? "Version published with MFA"
+  end
+
+  test "shows 'version' block only if latest does not require MFA but viewed version does" do
+    @version.update_attribute :metadata, { "rubygems_mfa_required" => "false" }
+    create(:version, :mfa_required, rubygem: @rubygem, number: "0.1.1")
+
+    visit rubygem_version_path(@rubygem, "0.1.1")
+
+    refute page.has_content? "New versions require MFA"
+    assert page.has_content? "Version published with MFA"
+  end
 
+  test "does not show either block if neither latest or viewed version require MFA" do
+    @version.update_attribute :metadata, { "rubygems_mfa_required" => "false" }
+    create(:version, rubygem: @rubygem, number: "0.1.1")
+
+    visit rubygem_version_path(@rubygem, "0.1.1")
+
+    refute page.has_content? "New versions require MFA"
+    refute page.has_content? "Version published with MFA"
+  end
+
+  test "shows both blocks if MFA enabled for latest version and viewing latest version" do
+    @version.update_attribute :metadata, { "rubygems_mfa_required" => "true" }
+    
     visit rubygem_path(@rubygem)
+    
+    assert page.has_content? "New versions require MFA"
+    assert page.has_content? "Version published with MFA"
+  end
 
-    assert page.has_content? "true"
+  test "shows neither block if MFA disabled for latest version and viewing latest version" do
+    @version.update_attribute :metadata, { "rubygems_mfa_required" => "false" }
+    
+    visit rubygem_path(@rubygem)
+    
+    refute page.has_content? "New versions require MFA"
+    refute page.has_content? "Version published with MFA"
```

Comment:
Is there anything new being tested in this test vs in [this one](https://github.com/rubygems/rubygems.org/pull/3079/files#diff-686870ebce0171eae254d0d41a3197e5fca504c025eff34b6d39fd74bc794d4bR188-R196)? It feels a bit redundant

Reply:
The difference with the last two tests is that they are looking at the gem's main page instead of the specific version of a gem. Do you think they are worth keeping then?

Reply:
I guess the tests would catch if anyone ever changed one of those to not use the template for some reason